pkg/api/api.go

```diff
@@ -54,5 +57,20 @@ func NewClient(baseURL string, opts ...Option) *Client {
 // Do executes c.doFunc(), which in turn allows wrapping c.client.Do() and manipulating
 // the request behavior of the api client.
 func (c *Client) Do(req *http.Request) (*http.Response, error) {
-	return c.doFunc(c, req)
+	resp, err := c.doFunc(c, req)
+	if err != nil {
+		if errors.Is(err, syscall.ECONNREFUSED) {
+			log.Warnf("connection refused: %s. It will retry with cloudfare dns", req.URL)
```

Comment:
Change to:

`log.Warnf("dns lookup failed: %s. Retrying with fallback dns resolver", req.URL)`